File: Go/jobspls/algorithms/cyclic.go
package algorithms

// Rotate an array 
func Rotate[T any](values []T, by int) []T {
	shifted := make([]T, len(values))

	for i, v := range values {
		j := (i + by) % len(values)
		shifted[j] = v
	}

	return shifted
}

File: Python/jobspls/algorithms/cyclic.py
from typing import Any

"""Rotate an array."""
def rotate(array: list[Any], by: int) -> list[Any]:
    rotated: list[Any] = [None] * len(array)

    for i, value in enumerate(rotated):
        j: int = (i + by) % len(array)
        rotated[j] = value

    return rotated
